app/config/FeatureSwitches.scala

```diff
@@ -16,24 +16,45 @@
 
 package config
 
+import com.google.inject.ImplementedBy
 import org.apache.commons.lang3.BooleanUtils
 import play.api.Configuration
 import play.api.mvc.Request
 
-case class FeatureSwitches(featureSwitchConfig: Configuration) {
+import javax.inject.{Inject, Singleton}
+
+@ImplementedBy(classOf[FeatureSwitchesImpl])
+trait FeatureSwitches {
+
+  def isCl402Enabled: Boolean
+  def isTemporalValidationEnabled(implicit request: Request[_]): Boolean
+}
+
+@Singleton
+class FeatureSwitchesImpl(featureSwitchConfig: Configuration) extends FeatureSwitches {
+
+  @Inject
+  def this(appConfig: AppConfig) = this(appConfig.featureSwitches)
+
+  val isCl402Enabled: Boolean = isEnabled("cl402")
 
   def isTemporalValidationEnabled(implicit request: Request[_]): Boolean = {
-    if (isEnabled("allowTemporalValidationSuspension.enabled")) {
+    if (isConfigTrue("allowTemporalValidationSuspension.enabled")) {
```

Comment:
To save adding ".enabled" to the key name in two places:
```suggestion
    if (isEnabled("allowTemporalValidationSuspension")) {
```